### PR TITLE
Converted the SARS-CoV-2 map to use the `PathogenMap` component.

### PR DESCRIPTION
**Pull request by SeanKennyNF**
Resolves https://github.com/PathoTracker/Pathotracker/issues/64 and also resolves https://github.com/PathoTracker/Pathotracker/issues/35 for the SARS-CoV-2 map since this bug was also present there.

The SARSCoV2 stuff is kind of buggy but this shouldn't introduce any new problems at least hahaha. 

R.I.P. `useMap` 2023-2023

pr-71-demo

### CODE REVIEW

**Review by EliasPinno — CHANGES_REQUESTED** [review]
I think we should deliberately not merge this before Heidelberg. Just no reason to do so if since I feel like it possible this could have side effects for arbotracker.

**Review by EliasPinno — APPROVED** [review]

**Comment by SeanKennyNF** [comment]
Will merge once mapbox API key is working again.

**Pull request merged**
